kinesis_video_streamer/launch/kinesis_video_streamer.launch.py

```diff
@@ -59,14 +63,49 @@ def generate_launch_description():
     launch.actions.DeclareLaunchArgument(
       CONFIG,
       default_value=default_config
+    ),
+    launch.actions.DeclareLaunchArgument(
+      AWS_REGION,
+      default_value=default_aws_region
+    ),
+    launch.actions.DeclareLaunchArgument(
+      STREAM_NAME,
+      default_value=default_stream_name
+    ),
+    launch.actions.DeclareLaunchArgument(
+      REKOGNITION_DATA_STREAM,
+      default_value=default_rekognition
     )
-   ])
+  ])
+
+  node_parameters = [launch.substitutions.LaunchConfiguration(CONFIG)]
+  node_parameters.append({
+    'aws_client_configuration': {
+      'region': launch.substitutions.LaunchConfiguration(AWS_REGION)
+    }
+  })
+  node_parameters.append({
+    'kinesis_video': {
+      'stream0': {
+        'stream_name': launch.substitutions.LaunchConfiguration(STREAM_NAME)
+      }
+    }
+  })
+  node_parameters.append({
+    'kinesis_video': {
+      'stream0': {
+        'rekognition_data_stream': launch.substitutions.LaunchConfiguration(REKOGNITION_DATA_STREAM)
+      }
+    }
+  })
+  logger_config_path = get_logger_config_path(default_config_yaml)
+  if logger_config_path is not None:
+    node_parameters.append({
+      'kinesis_video': {
+        'log4cplus_config': logger_config_path
+      }
+    })
 
-  logger_config_path = get_logger_config_path()
```

Comment:
FYI, the kind of logic under this function call means users cannot override the `log4cplus_config` under their own configuration YAML file, since this function only looks at the `log4cplus_config` defined under `sample_config.yaml`.

Reply:
(This PR does not solve this problem.)

Reply:
Unsure if it's possible to add DeclareLaunchArgument inside an if statement but if you can then you could add it here with a name of `log4cplus_config` and default value of `logger_config_path` to give some way for the user to overwrite it. Then in this `node_parameters.append` section specify it with LaunchConfiguration.

Reply:
Yeah, overriding `log4cplus_config` via a launch parameter is clearly possible, but it is not entirely clear if overriding it via a configuration YAML file is at all possible, because it requires loading the user's custom configuration YAML file (which is only known at launch time, not at `generate_launch_description()` time) and then subsequently calling `get_logger_config_path()` (which again then must be at launch time).

Reply:
Yea the user would need to do the same method of reading in their config file and passing the param from it like we do in the sample applications (https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/blob/ros2/robot_ws/src/cloudwatch_robot/launch/monitoring.launch.py#L28)